### ISSUE TITLE
TextButtons without text

### ISSUE THREAD

**Issue by Zweistein2** [mini2Dx/mini2Dx#70]
Heyho,

the last days I stumbled across a small bug (or my fault). If I load a modal like this:

```java
uiContainer = new UiContainer(gc, assetManager);

try
{
    AlignedModal modal = Mdx.xml.fromXml(Gdx.files.internal("uis/MainMenu.xml").reader(), AlignedModal.class);
    TextButton newGameButton = (TextButton) modal.getElementById("newGameButton");
    TextButton loadGameButton = (TextButton) modal.getElementById("loadGameButton");
    TextButton optionsButton = (TextButton) modal.getElementById("optionsButton");
    TextButton exitButton = (TextButton) modal.getElementById("exitButton");
    newGameButton.addActionListener(createActionListener(GameMenuScreen.ID));
    loadGameButton.addActionListener(createActionListener(GameMenuScreen.ID));
    optionsButton.addActionListener(createActionListener(ConfigScreen.ID));
    exitButton.addActionListener(new ActionListener() {
         @Override
         public void onActionBegin(ActionEvent event) { Gdx.app.exit(); }

         @Override
         public void onActionEnd(ActionEvent event) {

         }
    });

    uiContainer.add(modal);
} catch (SerializationException e)
{
    e.printStackTrace();
}

mainStage = gc.createStage(new FitViewport(viewWidth, viewHeight));
uiStage = gc.createStage(new FitViewport(viewWidth, viewHeight));
```

Everything is fine except for the last TextButton (exitButton) as it has no text... If I add another TextButton the exitButton shows its text (without changing its code), but the newly added one doesn't. It does function though (If I click it, the game stops).

Now the question... is it a bug or have I forgotten something?

**Comment by tomcashman** [comment]
Hmmm this might be a bug but haven't encountered it in my own games. Can you show me the contents of uis/MainMenu.xml?

**Comment by Zweistein2** [comment]
```
<?xml version="1.0"?>
<data id="MainMenuModal">
    <visibility>VISIBLE</visibility>
    <verticalAlignment>MIDDLE</verticalAlignment>
    <horizontalAlignment>CENTER</horizontalAlignment>
    <layout>xs-4c sm-4c md-4c lg-4c xl-3c</layout>
    <children length="4">
        <value class="org.mini2Dx.ui.element.Row" id="newGameRow">
            <visibility>VISIBLE</visibility>
            <children length="1">
                <value class="org.mini2Dx.ui.element.TextButton" id="newGameButton">
                    <visibility>VISIBLE</visibility>
                    <children length="1">
                        <value class="org.mini2Dx.ui.element.Label">
                            <visibility>VISIBLE</visibility>
                            <text>Spiel Starten</text>
                            <horizontalAlignment>CENTER</horizontalAlignment>
                        </value>
                    </children>
                </value>
            </children>
        </value>
        <value class="org.mini2Dx.ui.element.Row" id="loadGameRow">
            <visibility>VISIBLE</visibility>
            <children length="1">
                <value class="org.mini2Dx.ui.element.TextButton" id="loadGameButton">
                    <visibility>VISIBLE</visibility>
                    <children length="1">
                        <value class="org.mini2Dx.ui.element.Label">
                            <visibility>VISIBLE</visibility>
                            <text>Spiel laden</text>
                            <horizontalAlignment>CENTER</horizontalAlignment>
                        </value>
                    </children>
                </value>
            </children>
        </value>
        <value class="org.mini2Dx.ui.element.Row" id="optionsRow">
            <visibility>VISIBLE</visibility>
            <children length="1">
                <value class="org.mini2Dx.ui.element.TextButton" id="optionsButton">
                    <visibility>VISIBLE</visibility>
                    <children length="1">
                        <value class="org.mini2Dx.ui.element.Label">
                            <visibility>VISIBLE</visibility>
                            <text>Optionen</text>
                            <horizontalAlignment>CENTER</horizontalAlignment>
                        </value>
                    </children>
                </value>
            </children>
        </value>
        <value class="org.mini2Dx.ui.element.Row" id="exitRow">
            <visibility>VISIBLE</visibility>
            <children length="1">
                <value class="org.mini2Dx.ui.element.TextButton" id="exitButton">
                    <visibility>VISIBLE</visibility>
                    <children length="1">
                        <value class="org.mini2Dx.ui.element.Label">
                            <visibility>VISIBLE</visibility>
                            <text>Beenden</text>
                            <horizontalAlignment>CENTER</horizontalAlignment>
                        </value>
                    </children>
                </value>
            </children>
        </value>
    </children>
</data>
```

**Comment by tomcashman** [comment]
Nothing irregular there. I'll try reproduce this evening.

**Comment by tomcashman** [comment]
I couldn't reproduce this last night using version 1.5.2. Which version of mini2Dx are you using?

This is the code I used to load your XML file:

```java
import org.mini2Dx.core.Mdx;
import org.mini2Dx.core.assets.FallbackFileHandleResolver;
import org.mini2Dx.core.game.BasicGame;
import org.mini2Dx.core.graphics.Graphics;
import org.mini2Dx.core.serialization.SerializationException;
import org.mini2Dx.ui.UiContainer;
import org.mini2Dx.ui.UiThemeLoader;
import org.mini2Dx.ui.element.AlignedModal;
import org.mini2Dx.ui.element.TextButton;
import org.mini2Dx.ui.style.UiTheme;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.assets.loaders.FileHandleResolver;
import com.badlogic.gdx.assets.loaders.resolvers.ClasspathFileHandleResolver;
import com.badlogic.gdx.assets.loaders.resolvers.InternalFileHandleResolver;
import com.badlogic.gdx.graphics.Texture;

public class MyMini2DxGame extends BasicGame {
	public static final String GAME_IDENTIFIER = "com.mystudio.gamename";

	private AssetManager assetManager;
	private UiContainer uiContainer;
	
	@Override
    public void initialise() {
		FileHandleResolver fileHandleResolver = new FallbackFileHandleResolver(new ClasspathFileHandleResolver(), new InternalFileHandleResolver());
		assetManager = new AssetManager(fileHandleResolver);
		//Add mini2Dx-ui theme loader
        assetManager.setLoader(UiTheme.class, new UiThemeLoader(fileHandleResolver));

        //Load default theme
        assetManager.load(UiTheme.DEFAULT_THEME_FILENAME, UiTheme.class);
		
		uiContainer = new UiContainer(this, assetManager);
		
		try
		{
		    AlignedModal modal = Mdx.xml.fromXml(Gdx.files.internal("ui.xml").reader(), AlignedModal.class);
		    TextButton newGameButton = (TextButton) modal.getElementById("newGameButton");
		    TextButton loadGameButton = (TextButton) modal.getElementById("loadGameButton");
		    TextButton optionsButton = (TextButton) modal.getElementById("optionsButton");
		    TextButton exitButton = (TextButton) modal.getElementById("exitButton");

		    uiContainer.add(modal);
		} catch (SerializationException e)
		{
		    e.printStackTrace();
		}
    }
    
    @Override
    public void update(float delta) {
    	if(!assetManager.update()) {
    		return;
    	}
    	if(!uiContainer.isThemeApplied()) {
    		uiContainer.setTheme(assetManager.get(UiTheme.DEFAULT_THEME_FILENAME, UiTheme.class));
    	}
    	uiContainer.update(delta);
    }
    
    @Override
    public void interpolate(float alpha) {
    	uiContainer.interpolate(alpha);
    }
    
    @Override
    public void render(Graphics g) {
		uiContainer.render(g);
    }
}
```

**Comment by Zweistein2** [comment]
I'm using mini2dx 1.5.2 and very much the same code to load the xml-file. Maybe there's a problem when used with stages?

```java
mainStage = gc.createStage(new FitViewport(viewWidth, viewHeight));
uiStage = gc.createStage(new FitViewport(viewWidth, viewHeight));
```

I'll try it out later without them.

**Comment by Zweistein2** [comment]
If I use stages, the buttontext won't be rendered, if I don't use stages, the button is rendered absolutely perfect.

**Comment by tomcashman** [comment]
Ok I will investigate.

**Comment by tomcashman** [comment]
The latest commit should fix the issue but I'll need you to confirm. Can you clone this repo and run ```./gradlew clean install``` then set your mini2Dx version to 1.9.3-SNAPSHOT?

I noticed a lot of issues in using the stage2D. You would have had to create your own spritebatch in your code? This was incorrect and should be removed - the spritebatch from the Graphics class will be used now when gc.createStage is called. Also note that your should use ```g.drawStage(uiStage);``` instead of ```uiStage.draw()``` for this to work correctly.

**Comment by Zweistein2** [comment]
It's working perfectly now. Thank you very much!

**Comment by tomcashman** [comment]
Awesome! I'll tag a new version this evening and upload to Maven Central.

**Comment by tomcashman** [comment]
Fixed by commit f0b415a324d262e716ec76ed309d105b564305d9.

Fixed in release 1.5.3.
